app/assets/javascripts/user_analytics.js

```diff
@@ -0,0 +1,185 @@
+    if (element === syncNudgeCardElement()) {
+      return parseInt(position) + 1;
+    }
+  }
+
+  return -1
+}
+
+//
+// behaviour
+//
+function showSyncNudge(currentSlide) {
+  var weHaveDataForToday = false;
+
+  for (let date of dailyStatisticsDates()) {
+    if (date === formatDate(today())) {
+      // don't count the sync nudge card as a real day card
+      weHaveDataForToday = true;
+      syncNudgeCardElement().classList.remove("day");
+      break;
+    }
+  }
+
+  if (!weHaveDataForToday && (currentSlide === syncNudgeCardPosition())) {
+
+    // re-enable the nudge card if we don't have today's data
+    syncNudgeCardElement().classList.add("day");
+    syncNudgeCardElement().style.display = 'block';
+  }
+}
+
+function updateDateAtEndOfCarousel() {
+  var latestDate = formatDate(latestDateInDailyStatistics());
+  var todayDate = formatDate(today());
+  var yesterdayDate = formatDate(yesterday());
+  var endOfCarouselElement = allDaysInCarouselElements()[0].querySelector('.stat-day');
+
+  if (latestDate === todayDate || latestDate === yesterdayDate) {
+    endOfCarouselElement.innerHTML = formattedTodayString();
+    return;
+  }
+
+  if (latestDate < yesterdayDate) {
+    endOfCarouselElement.innerHTML = formattedTomorrowDate();
+  }
+}
+
+function showDailyProgressCards(next) {
+  var elementsForAllDays = allDaysInCarouselElements();
+  var nextButton = nextSlideButtonElement();
+  var prevButton = prevSlideButtonElement();
+
+  nextButton.disabled = false;
+  prevButton.disabled = false;
+
+  if (next === elementsForAllDays.length) {
+    nextButton.disabled = true;
+  }
+
+  if (next === 1) {
+    prevButton.disabled = true;
+  }
+
+  for (let i = 0; i < elementsForAllDays.length; i++) {
+    elementsForAllDays[i].classList.remove("day-show");
+  }
+
+  elementsForAllDays[window.lastSlidePositionForProgressCards - 1].classList.add("day-show");
+}
+
+function refreshCarousel(slidePosition) {
+  showSyncNudge(slidePosition);
+  showDailyProgressCards(slidePosition);
+  updateDateAtEndOfCarousel()
+}
+
+//
+// loads at page refresh
+//
+window.onload = function () {
+  window.lastSlidePositionForProgressCards = 1;
+  refreshCarousel(window.lastSlidePositionForProgressCards)
+};
+
+//
+// on-click events
+//
+function nextSlide(increment) {
+  refreshCarousel(window.lastSlidePositionForProgressCards += increment)
+}
+
+function filterDataByGender(tableName) {
+  var tableElements = document.getElementsByClassName('progress-table ' + tableName);
+  var tableFilterElement = document.getElementsByClassName('card-dropdown ' + tableName);
+  var selectedOption = tableFilterElement[0].selectedOptions[0].value;
+  var selectedTableElement = document.getElementsByClassName('progress-table ' + tableName + ' ' + selectedOption);
+
+  for (let i = 0; i < tableElements.length; i++) {
+    tableElements[i].style.display = 'none';
+  }
+
+  selectedTableElement[0].style.display = 'inline-table';
```

Comment:
This hides all first, then shows the selected one, is that right?

Reply:
That's correct.